modules/arrow/src/lib/parse-arrow-sync.js

```diff
@@ -13,9 +26,76 @@ export default function parseArrowSync(arrayBuffer, options) {
   arrowTable.schema.fields.forEach(field => {
     // This (is intended to) coalesce all record batches into a single typed array
     const arrowColumn = arrowTable.getColumn(field.name);
-    const values = arrowColumn.toArray();
+    const metadata = arrowColumn.metadata;
+    let values;
+    if (metadata.has('hex64str') && metadata.get('hex64str') && arrowColumn.name === 'h3') {
```

Comment:
* I'd suggest making this a loader option whether to use the 'hex64str' column metadata or not, since this is not a standard
* Please remove the check for the column name

Reply:
I don't now rely on metadata.has('hex64str') anymore. Only 64-bit data parsing for now